arch/x86/adler32_avx.c

```diff
@@ -14,6 +14,21 @@
 
 #ifdef X86_AVX2_ADLER32
 
+/* 64 bit horizontal sum, adapted from Agner Fog's vector library. */
+static inline uint64_t hsum(__m256i x) {
+    __m256i sum1 = _mm256_shuffle_epi32(x, 0x0E);
+    __m256i sum2 = _mm256_add_epi64(x, sum1);
+    __m128i sum3 = _mm256_extracti128_si256(sum2, 1);
+#ifdef __X86_64__
```

Comment:
``__x86_64__``, not ``__X86_64__``, and ``__x86_64__ `` is undefined on Microsoft Visual Studio.
Use
```c++
#if defined(__x86_64__) || defined(_M_X64)
```

like this:
https://github.com/zlib-ng/zlib-ng/blob/f77af71e77cdb9d41d5e3a606e2a8dd67034e387/cmake/detect-arch.c#L7